src/api/search/test/query.test.js

```diff
@@ -1,9 +1,230 @@
+        {
+          value: '06/04/2020',
+          msg: 'invalid date format',
+          param: 'to',
+          location: 'query',
+        },
+      ]);
+    });
+
+    it('Return error 400 if "from" param has an invalid date format', async () => {
+      let res = await request(app)
+        .get('/advanced')
+        .query({ post: 'ElasticSearch', from: 'foo-bar 2020' });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toStrictEqual('foo-bar 2020');
+
+      res = await request(app)
+        .get('/advanced')
+        .query({ post: 'ElasticSearch', from: '2020-25-23' });
+      expect(res.status).toBe(400);
+      expect(res.body).toStrictEqual([
+        {
+          value: '2020-25-23',
+          msg: 'invalid date format',
+          param: 'from',
+          location: 'query',
+        },
+      ]);
+    });
+
+    it('Return error 400 if "perPage" param is not an integer between 1 to 10', async () => {
+      let res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 11 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('11');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 0 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('0');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 1.5 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('1.5');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', perPage: 'one' });
+      expect(res.status).toBe(400);
+      expect(res.body).toStrictEqual([
+        {
+          value: 'one',
+          msg: 'perPage should be empty or a number between 1 to 10',
+          param: 'perPage',
+          location: 'query',
+        },
+      ]);
+    });
+
+    it('Return error 400 if "page" param is not an integer between 0 to 999', async () => {
+      let res = await request(app).get('/advanced').query({ post: 'ElasticSearch', page: 1101 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('1101');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', page: -1 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('-1');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', page: 55.5 });
+      expect(res.status).toBe(400);
+      expect(res.body[0].value).toBe('55.5');
+
+      res = await request(app).get('/advanced').query({ post: 'ElasticSearch', page: 'five' });
+      expect(res.status).toBe(400);
+      expect(res.body).toStrictEqual([
+        {
+          value: 'five',
+          msg: 'page should be empty or a number between 0 to 999',
+          param: 'page',
+          location: 'query',
+        },
+      ]);
+    });
+  });
+
+  describe('Test queries that pass validation', () => {
+    const { mock } = Elastic();
```

Comment:
It might be worth throwing a comment above this line to indicate that all Satellite Elastic clients will share this mock, and to link readers to the docs in Satellite.